core/src/main/java/ai/timefold/solver/core/impl/heuristic/selector/move/generic/list/kopt/MultipleDelegateList.java

```diff
@@ -70,7 +70,7 @@ public void applyChangesFromCopy(MultipleDelegateList<?> copy) {
 
     public int getIndexOfValue(ListVariableStateSupply<?> listVariableStateSupply, Object value) {
         var elementLocation = listVariableStateSupply.getLocationInList(value);
-        if (elementLocation instanceof LocationInList elementLocationInList) {
+        if (elementLocation instanceof LocationInList<?> elementLocationInList) {
```

Comment:
This is kind of casting operation we can avoid by changing the `ListVariableStateSupply` contract.

See my suggestions related to `LocationInList`.

Reply:
Answered above.